iotedgedev/modules.py

```diff
@@ -138,9 +138,17 @@ def build_push(self, no_build=False, no_push=False):
                 if not no_push:
                     # PUSH TO CONTAINER REGISTRY
                     self.output.info("PUSHING DOCKER IMAGE: " + tag)
+                    registry_key = None
+                    for key, registry in self.envvars.CONTAINER_REGISTRY_MAP.items():
+                        if registry.server.lower() == tag.split('/')[0].lower():
+                            registry_key = key
+                            break
+                    if registry_key is None:
+                        self.output.error("Could not find registry server with name {0}. Please make sure your envvar is set.".format(tag.split('/')[0].lower()))
+                    self.output.info("module json reading {0}".format(tag))
 
                     for line in self.dock.docker_client.images.push(repository=tag, stream=True, auth_config={
-                            "username": self.envvars.CONTAINER_REGISTRY_USERNAME, "password": self.envvars.CONTAINER_REGISTRY_PASSWORD}):
+                            "username": self.envvars.CONTAINER_REGISTRY_MAP[registry_key].username, "password": self.envvars.CONTAINER_REGISTRY_MAP[registry_key].password}):
```

Comment:
Since you are doing hte lookup twice, you should get the object once and then use the props on that object.
```
cr = self.envvars.CONTAINER_REGISTRY_MAP[key]
cr.username
cr.password
```